### DISCUSSION TITLE
Inline longform content

### DISCUSSION THREAD

**Discussion by ishetnogferre** [craftcms/ckeditor#256] [Ideas] [answered]
Right now the craftEntryModel is a **$blockObject**. By also adding an **$inlineObject** option, we'd be able to re-use the logic and powerful functionalities of the nested entries in inline elements as well.

The current use case for us in a client who wants to select stores (entries) to mention them through their website content (e.g. blog posts) while the store owners can update their links themselves. So a normal link wouldn't suffice. We'd link the entry via a nested entry and output the link from the entry on the front-end. (We can also output a list of mentions by using entry queries for this entry type).

The chip shows the link text and the linked stored.

Quick little CSS changes to how that could look:
![Screenshot 2024-06-20 at 13 14 42](https://github.com/craftcms/ckeditor/assets/6649706/13178232-d609-40e5-bd7f-5ab4a16291a2)

Other possible use cases, I can think of would be:
- modals
- tooltips
- ...

**Comment by brandonkelly** [comment, marked as answer]
> The current use case for us in a client who wants to select stores (entries) to mention them through their website content (e.g. blog posts) while the store owners can update their links themselves. So a normal link wouldn't suffice.

If you link to the store entry via the “Link to an entry” option, the resulting URL will actually get stored as a [reference tag](https://craftcms.com/docs/5.x/system/reference-tags.html) for the selected entry’s URL (e.g. `{entry:123:url}`), rather than a hard-coded URL. So if the store entry’s URL changes, your CKEditor field content will pull in the change.

Regarding your suggestion, the chip UI you proposed looks confusing to me. As an author I would assume that it represented a relation to another entry titled “test - Borrenberghs”, and that the rendered field on the front end would replace it with a link to that entry (rather than representing a nested entry, with its own custom fields and purpose for the field).

**Reply by ishetnogferre** [reply]
Oh wow, didn't know you could also reference [custom fields on the elements](https://craftcms.com/docs/5.x/system/reference-tags.html#examples) 😮 I don't suppose there's a built-in way to get like a list of a certain entry and their references used in CKEditor? Because then we'd have solved our use case 😄 

EDIT: Oh, this seems to work: https://discord.com/channels/456442477667418113/456442867163070466/1095119460848894082

**Comment by infin-8-morphosis** [comment]
This is closed but I was actually looking for a way to do this. 

I wanted to use the entry partials feature to add tooltips to words / links [using this method](https://www.w3schools.com/css/css_tooltip.asp), but it seems to be impossible to have an entry in the middle of a paragraph. Which makes sense, for I'd imagine the majority of use cases.

Adding the html manually via source is very ugly / somewhat broken (specifically if you use 'tooltip' as the class), and while yes, it is sort of possible to use reference tags to do it, it makes sentences frustrating to read, even knowing what the reference tags are parsed as. And that's assuming the tooltip is just one reference, adding icons or links would become excruciating to manage.

Source-wise it's easy to stick entries in paragraphs but it, I guess corrects itself on leaving the source editing. I wouldn't mind it looking broken if it worked, if it's possible to stop it doing that also.
